### ISSUE TITLE
Localstack stops responding when it is sent some amount of requests concurrently

### ISSUE THREAD

**Issue by Veetaha** [localstack/localstack#2825]
# This is a bug report

# Detailed description

The title pretty much fully describes the bug. I believe this is not service-specific.
I have stumbled with this testing concurrency durability of my services that use dynamodb.
The bug is floating, but the probability to reproduce it increases with the number of requests sent concurrently.
I've seen it come up even with 3 concurrent requests, but it comes up with 10 ones more or less steadily.

## Expected behavior

Localstack should respond to any amount of concurrent requests. I would even recommend using `dynamodb-local` image for dynamodb implementation (if it turns out that errors are service-specific), because it works properly with any number of concurrent requests.

## Actual behavior

Sending some amount of concurrent requests to some AWS services might make the entire localstack unresponsive (see examples).

# Steps to reproduce

Just start localstack and try sending some 10+ requests to dynamodb or 30+ messages SQS concurrently.
[Here is a demo of how to do it with the example sdk code pasted further](https://drive.google.com/file/d/1ZET6Or-eDJdmmpNqDPveWbv5kknHf-eg/view?usp=sharing)

## Command used to start LocalStack

```
localstack start
```

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)

<details>
<summary> Javascript </summary>

To bootstrap run:

```
npm init --yes
npm i aws-sdk
node index.js
```

There are two scripts with sending concurrent requests to sqs/dynamodb, choose one of them setting `const main` variable.

```ts
// index.js

const aws = require("aws-sdk");

const CONCURRENT_REQUESTS = 10;
// set to sqsTest, to run test with sqs
const main = ddbTest;

const cfg = {
    region: "us-east-2",
    endpoint: "http://localhost:4566",
};

main().then(
    () => console.log("Done"),
    err => console.error("Whoops:", err)
);

// Test sending PUT requests to DynamoDB concurrently
async function ddbTest() {
    const ddb = new aws.DynamoDB(cfg);
    const TableName = "sandbox-table";

    // Create a sample table
    await ddb.createTable({
        TableName,
        AttributeDefinitions: [{ AttributeName: "partition_key", AttributeType: "S", }],
        KeySchema:            [{ AttributeName: "partition_key", KeyType: "HASH" }],
        BillingMode: "PAY_PER_REQUEST",
    }).promise();

    console.log("Table", TableName, "is created");

    // Make concurrent requests
    await waitForConcurrentRequests(i => {
        const Put = { Item: { partition_key: { S: `bar${i}` } }, TableName };
        return ddb.putItem(Put).promise();
    });

    // Check what was written to the table
    const result = await ddb.scan({ TableName }).promise();
    console.log("Items:", result.Items.length, result.Items);

    // Cleanup
    await ddb.deleteTable({ TableName }).promise();
}

async function waitForConcurrentRequests(makeRequest) {
    let requests = [];
    for (let i = 0; i < CONCURRENT_REQUESTS; ++i) {
        requests.push(makeRequest(i));
    }
    console.log("Awaiting for responses...");
    console.time("All requests");
    await Promise.all(requests).then(
        () => console.log("Ok"),
        err => console.error("Err", err)
    );
    console.timeEnd("All requests");
}


// Test sending messages to SQS concurrently
async function sqsTest() {
    const sqs = new aws.SQS(cfg);
    const { QueueUrl } = await sqs.createQueue({ QueueName: "test-queue" }).promise();

    await waitForConcurrentRequests(i => {
        return sqs.sendMessage({ MessageBody: `baz${i}`, QueueUrl }).promise();
    });

    // Pop all messages from the queue
    let allMessages = [];
    while (true) {
        const msgs = await sqs.receiveMessage({
            QueueUrl,
            MaxNumberOfMessages: 10
        }).promise();
        if (msgs.Messages === undefined) {
            break;
        }
        allMessages.push(...msgs.Messages.map(it => it.Body));
    }

    console.log("Messages:", allMessages.length, allMessages);

    // Cleanup
    await sqs.deleteQueue({ QueueUrl }).promise();
}
```
</details>

**Comment by whummer** [comment]
Thanks for reporting @Veetaha . We were able to reproduce this issue - it should be fixed in #2827. Can you please pull the Docker image and give it another try with `latest`? Tests with up to 80-90 parallel requests have now be run successfully. Hope that helps - please report here if the problem persists. Thanks

**Comment by Veetaha** [comment]
Yup, the issue seems to appear at 90-100+ concurrent dynamodb requests or 800-900+ sqs requests.
While this seems to be far enough beyond the adequate limits, I'd recommend handling the case of thread-pool exhaustion returning at least something status code from 500 plane (at the very least `500 Internal Error`, or [`429 Too Many Requests`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/429), though I am not sure the last one fits entirely)
